test-d/empty-object.ts

```diff
@@ -0,0 +1,20 @@
+import {expectAssignable, expectError, expectType} from 'tsd';
+import type {EmptyObject, IsEmptyObject} from '../index';
+
+declare let foo: EmptyObject;
+
+expectAssignable<{}>(foo);
+expectAssignable<{}>(foo = {});
+
+expectError(foo = []);
```

Comment:
```suggestion
expectError(foo = []);
expectError(foo = {x: 1});
```

Reply:
And can you add a test for the example in https://github.com/sindresorhus/type-fest/issues/395#issuecomment-1120609111?

Reply:
> And can you add a test for the example in [#395 (comment)](https://github.com/sindresorhus/type-fest/issues/395#issuecomment-1120609111)?

Isn't that the one? https://github.com/sindresorhus/type-fest/blob/ad34426f02a19845340a598cc77a80028cd0f327/test-d/empty-object.ts#L13

Reply:
Are we looking at the same example? I don't see how that is the same as one using a union:

```ts
type Union = EmptyObject | { id: string }; 

const a: Union = {}; 
const b: string = a.id // no error
const c = a.id; // inferred type: `never`;
```

Reply:
Good point ;)